### PR TITLE
fix: display error when an invalid DOI is detected during collection creation and upates 

### PR DESCRIPTION
**Pull request by Bento007**
- #2869

### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add DOI errors to the list of errors returned by the backend when attempting to modify a collection. 
- update the frontend to process DOI errors.

### CODE REVIEW

File: backend/corpora/lambdas/api/v1/collection.py [hunk 1]
```diff
@@ -162,16 +165,15 @@ def normalize_and_get_doi(body: dict, errors: list) -> Optional[str]:
...
     Retrieves publisher metadata from Crossref.
     """
-    provider = crossref_provider.CrossrefProvider()
+    provider = CrossrefProvider()
     try:
         return provider.fetch_metadata(doi)
     except CrossrefDOINotFoundException:
-        # TODO: add an error message
-        raise InvalidParametersHTTPException(detail="DOI cannot be found on Crossref")
+        errors.append({"name": "doi", "reason": "DOI cannot be found on Crossref"})
```
Comment (nayib-jose-gloria) [inline backend/corpora/lambdas/api/v1/collection.py:176]: You're passing the string "doi" to "name", did you mean to pass the string variable without the quotes?

Comment (Bento007) [reply backend/corpora/lambdas/api/v1/collection.py]: the error message has change.

**Comment by danieljhegeman** [comment]
@Bento007 has this been tested in rdev or dev?

**Review by danieljhegeman — APPROVED** [review]
This looks ok to me. May want a closer eye from FE crew on the ts changes, but again, nothing looks amiss.

Awaiting results of testing...

**Review by seve — APPROVED** [review]

**Pull request merged**